src/vs/workbench/contrib/chat/browser/chatContentParts/chatMarkdownContentPart.ts

```diff
@@ -149,15 +154,15 @@ export class ChatMarkdownContentPart extends Disposable implements IChatContentP
 						// TODO@joyceerhl: remove this code when we change the codeblockUri API to make the URI available synchronously
 						this.codeBlockModelCollection.update(codeBlockInfo.element.sessionId, codeBlockInfo.element, codeBlockInfo.codeBlockIndex, { text, languageId: codeBlockInfo.languageId, isComplete: isCodeBlockComplete }).then((e) => {
 							// Update the existing object's codemapperUri
-							this.codeblocks[codeBlockInfo.codeBlockIndex].codemapperUri = e.codemapperUri;
+							this.codeblocks[codeBlockInfo.codeBlockPartIndex].codemapperUri = e.codemapperUri;
```

Comment:
FYI @joyceerhl I think this is right, it didn't fail before because the edit session only had a single markdown part, now it can have multiple